src/main/java/com/whytowait/core/annotations/aspects/RequiresAuthoritiesAspect.java

```diff
@@ -0,0 +1,108 @@
+package com.whytowait.core.annotations.aspects;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.whytowait.api.common.exceptions.BadRequestException;
+import com.whytowait.api.common.exceptions.UnauthorizedException;
+import com.whytowait.config.BufferedRequestWrapper;
+import com.whytowait.core.annotations.RequiresAuthorities;
+import com.whytowait.core.annotations.enums.RequestSource;
+import com.whytowait.domain.models.enums.MerchantRole;
+import jakarta.servlet.http.HttpServletRequest;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Aspect
+@Component
+public class RequiresAuthoritiesAspect {
```

Comment:
Rename aspect class name to CheckMerchantManagerRoleAspect

Reply:
done